packs/chef/actions/lib/shellhelpers.py

```diff
@@ -43,17 +42,18 @@ def __init__(self, parser_options):
             self.parser.add_argument(k, lg, **kwargs)
             self._keyname[lg.lstrip('-')] = k
 
+
     def parse(self, argv=None):
-        argv = argv or sys.argv
-        return vars(self.parser.parse_args(args=argv))
+        return vars(self.parser.parse_args(argv))
+
 
     def short_arglist(self, kwargs=None):
-        kwargs = kwargs or {}
-        return self._arg_list(self.parse(), short=True)
+        return self._arg_list(kwargs or self.parse(), short=True)
+
 
     def long_arglist(self, kwargs=None):
-        kwargs = kwargs or {}
-        return self._arg_list(self.parse(), short=False)
+        return self._arg_list(kwargs or self.parse(), short=False)
```

Comment:
That's a generic function, maybe it will be reused. It doesn't matter anyway. If someone including me wants to build argument list using provided argument dict he is free to it.